client/src/Main.elm

```diff
@@ -1419,7 +1473,38 @@ update msg model =
             ( { model | vectorToAdd = vector }, Cmd.none )
 
         AddLevel0 newIns ->
-            ( model, model.api.save model.auth (Level0Vec newIns) )
+            ( model, model.api.save model.auth <| Level0Vec newIns )
+
+        AddLevel1 level1 ->
+            ( model, model.api.save model.auth <| LevelNVec level1 )
+
+        DownloadGenbankFile level1 ->
+            ( { model | vectorToAdd = Just (LevelNVec level1) }, model.api.genbank model.auth level1.id )
```

Comment:
Why is the `model` being changed here?

Reply:
vectorToAdd is by default Nothing. So if you click the download button, you'll try to change the content of "Nothing".